runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/PlayerComparisonOverlay.java

```diff
@@ -96,88 +102,78 @@ public Dimension render(Graphics2D graphics)
 			return null;
 		}
 
-		Actor opponent = opponentInfoPlugin.getOpponent();
+		Actor opponent = opponentInfoPlugin.getLastOpponent();
 
 		// Don't try to look up NPC names
 		if (!(opponent instanceof Player))
 		{
-			opponent = null;
+			return null;
```

Comment:
This was `opponent = null` previously so it would leave the table open until the 5s delay expired.